lib/LedgerSMB/Scripts/setup.pm

```diff
@@ -1111,7 +1112,10 @@ sub save_user {
            die $_;
        }
     };
-    return $duplicate if $duplicate;
+    if ( $duplicate ) {
+        $request->{dbh}->rollback;
```

Comment:
However, that's not the case here. What happens if we don't roll back here?

Reply:
1- A transaction is started and we return with it still opened.
2- The remainder of the code will assume no transaction and no commits will be done, for we rely on autocommitting.
3- The first error will invalidate everything from this point on

Reply:
Actually, I don't agree with (2). The request dispatcher will commit upon finalizing the request. So, commits *will* happen.

Reply:
And this will commit the first part of an incomplete transaction, much later in the process and at a completely unrelated moment.

Reply:
But you said the rest of the code won't commit. My point is that the code *will* commit.
In my view the moment of commit isn't unrelated as it is the commit to go with the returned document/user feedback.

My question was a different one though: is there code following this code which needs to query the database and needs to succeed at that?
If so, there are 2 follow up questions:
1. Why do we need a transaction here?
2. Why not just let the code run with auto-commit turned back on after the error?

Reply:
1- The code won't commit there. Somewhen, much after, an unrelated commit will impose the partial pending transaction.
2- Most likely we need a transaction because the original developer felt that employee and user creation were related. I agree with that assumption.
3- We create the employee, then the user. Without transaction, we would have to somehow undo the employee alterations.